### DISCUSSION TITLE
i18n Ignoring Routes

### DISCUSSION THREAD

**Discussion by batuhanbilginn** [vercel/next.js#28554] [Help]
Hey guys, I got some routes and pages that I don't want to translate. How can I ignore those routes? I've found a solution like below but seems like it doesn't work for some reason.

```js
module.exports = {
  i18n: {
    locales: ["tr", "en", "ru"],
    defaultLocale: "tr",
    ignoreRoutes: ["/example-path/", "/public/"],
  },
  images: {
    domains: ["localhost"],
  },
};
```

Thanks for your support!

**Comment by DCzajkowski** [comment]
As you can see in the source code of Next.js, `ignoreRoutes` [is not a valid configuration key](https://github.com/vercel/next.js/blob/e2506411a1cf287326082d2c326e58e8cc27a7a9/packages/next/server/config.ts#L380-L511). Unfortunately, there is no way to do this :/ I really need this as well

**Reply by batuhanbilginn** [reply]
I think it’s now possible with middleware. You can just create one in the page folder to redirect local homepage ;)

**Reply by DCzajkowski** [reply]
Interesting! I'll definitely take a look at this. Thanks!

**Reply by DCzajkowski** [reply]
Indeed that have done the trick. I've added a simple `.startsWith` `if` check with redirect. 👍

**Reply by idrisakmal** [reply]
hey [DCzajkowski](https://github.com/DCzajkowski) can you share how it was done using middleware solution?

**Reply by DCzajkowski** [reply]
Unfortunately, we scraped the whole idea. Seemed like too much work and effort to get satisfactory results.

Currently, we are translating all the pages.

**Reply by Qrau** [reply]
#### if you are trying to use the API calls, this solved my problem, hope it solves yours :)

- use next useRouter to get the current active language :\
    `import { useRouter } from  "next/router";`\
    `const router = useRouter();`\
    `const active_language = router.locale;`
    ###
- use window object to get the current full url of your app :\
    `const window_url = window.location.href;` 
   > **Note:**  Next.js executes code **first server-side**, then client-side (which includes the window object). 
    there are many ways to achieve this, if you get "Window is not defined" read [this article](https://stackoverflow.com/questions/55151041/window-is-not-defined-in-next-js-react-app).
    ###
    
- set your api url based on the current active language,\
assuming your default language is "en" :\
   `let api_url = "";`\
   `if(active_language === "en") {`\
   ``api_url = `${window_url.substr(0, window_url.lastIndexOf("/"))}/`};``\
   `console.log(window_url) // http://localhost:3000/shop`\
   `console.log(api_url) // http://localhost:3000/`\
   `};`\
   and your secondary language is "ar" :\
   `if(active_language === "ar") {`\
   `api_url = window_url.substr(0, window_url.lastIndexOf("/") - 2)};`\
   `console.log(window_url) // http://localhost:3000/ar/shop`\
   `console.log(api_url) // http://localhost:3000/`\
   `};`
   ###
- now your api url is set to the same url even if you switch the locale language :\
   ``fetch(`${api_url}/api/hello`});``

**Comment by subodhk01** [comment]
Did anyone find a solution for this?

**Comment by jeanhdev** [comment]
Needs this too for SEO related issues that might occur

**Comment by gaodeng** [comment]
any news?

**Comment by jr-rc** [comment]
Anyone planning to do this?

**Comment by MateusRanzani** [comment]
Hello! In my project, I applied translation using i18n. However, not all pages were meant to have translations, and when I started the deployment, it would throw an error.

I'm not sure if the solution I found for my issue would be feasible for you, but I resolved it using Next.js 'redirects' feature, as follows:

`next.config.js`
```
async redirects() {
      return [
         {
            source: '/es/project/page',
            destination: '/project/page',
            permanent: false,
          },
      ]
}
```

The routes for which there is no translation, I simply redirected to the original URL, which doesn't have a translation, and added permanent: false because they will have the translation in the future.

I hope this can help you in some way

**Reply by skylinkeming** [reply]
I've tried it, but it didn't work. Can somebody help me?

=========================
I found it, you have to add locale:false to make it work.
async redirects() {
      return [
         {
            source: '/es/project/page',
            destination: '/project/page',
            permanent: false,
            locale:false
          },
      ]
}
